awx/api/permissions.py

```diff
@@ -254,3 +265,16 @@ def has_object_permission(self, request, view, obj):
 class WebhookKeyPermission(permissions.BasePermission):
     def has_object_permission(self, request, view, obj):
         return request.user.can_access(view.model, 'admin', obj, request.data)
+
+
+class CredentialTypeExternalTestPermission(permissions.BasePermission):
+    def has_object_permission(self, request, view, obj):
+        return request.user.can_access(view.model, 'admin', obj, request.data)
```

Comment:
Since it's credential type, wouldn't this be strictly limited to superusers? The point I'm getting to there is that I'd like to not have a new permission class for this, because something that enforces this kind of permission surely already exists.

Reply:
Good point. I will remove it.